### ISSUE TITLE
WDT reset when sending websocket message on tick

### ISSUE THREAD

**Issue by pandadom** [esp8266/Arduino#3683]
#### Hardware
Hardware:			NodeMCU 1.0
Core Version:      	?2.1.0-rc2?

### Description

Problem description

Soft WDT reset occurs when sending a message on tick using websockets

### Settings in IDE

Module:  			NodeMCU 1.0
Flash Size: 		4MB
CPU Frequency:		80Mhz
Flash Mode:			?qio?
Flash Frequency:	?40Mhz?
Upload Using:		?OTA / SERIAL?
Reset Method:		?ck / nodemcu?

### Sketch

#include <Arduino.h>
#include <ESP8266WiFi.h>
#include <ESP8266WiFiMulti.h>
#include <WebSocketsServer.h>
#include <Hash.h>
#include <ESP8266HTTPClient.h>
#include <ESP8266httpUpdate.h>
#include <Ticker.h>
#include "EspSaveCrash.h"

////////////////config////////////////
WebSocketsServer webSocket = WebSocketsServer(8000);

const char* ssid     = "3neo_2.4Ghz_E3CC"; //wifi credentials
const char* password = "";// wifi credentials

#define USE_SERIAL Serial
ESP8266WiFiMulti WiFiMulti; // wifi connector
uint8_t wsNum = 0;//websocket client num (i assume its the ip)

IPAddress mcuIP; // ip received from wifi connector

bool stopSending = false; //
Ticker tickMeasure; // ticker used to measure ... like tmr in lua

void sayLine(String txt, bool line = true) {
  //function that prints string to serial output etc
  if (line) {
    Serial.println(txt);
  }
  else {
    Serial.print(txt);
  }
}

void measure(int num) {

  webSocket.sendTXT(num, "asdf");// THIS CAUSES THE RESET
}

void webSocketEvent(uint8_t num, WStype_t type, uint8_t * payload, size_t lenght) {

  switch (type) {
    case WStype_DISCONNECTED:
      {
        stopSending = true;
        tickMeasure.detach();
        sayLine("----------");
        sayLine(String(num));
        sayLine("disconnected");
      }
      break;

    case WStype_CONNECTED:
      {
        IPAddress ip = webSocket.remoteIP(num);
        sayLine("----------");
        sayLine(String(ip));
        sayLine("connected");
      }
      break;

    case WStype_TEXT:
      {

        String text = String((char *) &payload[0]);
        sayLine("----------");
        sayLine("received: ");
        sayLine(text, false);
        sayLine(String(num));

        if (text == "measure") {

          tickMeasure.attach(1, measure, int(num)); // call every second

        }
        break;

        if (text == "stopSending") {

          stopSending = true;
          tickMeasure.detach();
        }
      }
      break;
  }
}

void connectWifi() {

  WiFi.begin(ssid, password);
  sayLine("Connecting ...");
  while (WiFi.status() != WL_CONNECTED) {
    delay(1000);
    sayLine(".", false);
  }
  sayLine("wifi connection established");
  sayLine("MCU ip address: ");
  mcuIP = WiFi.localIP();
  Serial.println(WiFi.localIP());
}

void setup() {
  Serial.begin (115200);
  SaveCrash.print();
  connectWifi();

  /////////set-up Web Socket////////////
  webSocket.begin();
  webSocket.onEvent(webSocketEvent);
  sayLine("=======set-up complete========");
}

void loop() {
  webSocket.loop();// handle web socket events --> see function above
}

### Debug Messages

Error Message:
<<<stack<<<

 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v09f0c112
~ld
⸮Crash information recovered from EEPROM
Crash # 1 at 24391 ms
Reason of restart: 3
Exception cause: 4
epc1=0x40206448 epc2=0x00000000 epc3=0x00000000 excvaddr=0x00000000 depc=0x00000000
>>>stack>>>
3ffffe70: 3ffeeee8 3fff0d20 00000000 402010a3 
3ffffe80: 00000000 3fff0d20 00000000 40203d40 
3ffffe90: 000057c4 00000000 0000001f 00000002 
3ffffea0: 00000004 00000000 3ffe8428 00000002 
3ffffeb0: 00000004 00000006 3fff0d14 40203a01 
3ffffec0: 00000000 00000000 00000000 00000000 
3ffffed0: 00000000 00000000 00000001 3ffef470 
3ffffee0: 00000001 3ffeeeb0 3ffeeee8 00000000 
3ffffef0: 00000001 00000001 3fff0d14 00000000 
3fffff00: 40227f80 00000000 0000007d 00000004 
3fffff10: 00000000 3ffe8428 3ffeeeb0 402044d2 
3fffff20: 00000001 00000000 006af40d 40227ee1 
3fffff30: 3ffeeee8 3fff0990 006af40d 40227fa4 
3fffff40: 401051ee 006af40a 60000600 60000600 
3fffff50: 0156d547 3ffee870 3fff0d2c 402044f7 
3fffff60: 4021dc40 3fff0d2c 3ffee870 4020214d 
3fffff70: 60000600 3fff0d2c 3ffee870 4021dc4d 
3fffff80: 4021dc92 3fffdab0 00000000 3fffdcb0 
3fffff90: 3ffee898 3fffdad0 3ffef45c 402060d3 
3fffffa0: 40000f49 40000f49 3fffdab0 40000f49

**Comment by igrr** [comment]
From http://arduino-esp8266.readthedocs.io/en/latest/libraries.html#ticker:

> It is currently not recommended to do blocking IO operations (network, serial, file) from Ticker callback functions. Instead, set a flag inside the ticker callback and check for that flag inside the loop function.
